src/controller/SetUpCodePairer.cpp

```diff
@@ -387,6 +403,11 @@ void SetUpCodePairer::OnBLEDiscoveryError(CHIP_ERROR err)
 #if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
 void SetUpCodePairer::OnDiscoveredDeviceOverWifiPAF()
 {
+    if (mWaitingForDiscovery[kWiFiPAFTransport] == false)
+    {
+        // Discovery has stopped or device has been discovered
```

Comment:
If discovery has stopped, touching mWaitingForDiscovery is potentially use-after-free.